src/core.ts

```diff
@@ -757,16 +753,16 @@ export class TemplateInstance {
   }
 
   update(values: any[]) {
-    let valueIndex = 0;
+    let i = 0;
+    for (const part of this._parts) {
+      if (part !== undefined) {
+        part.setValue(values[i]);
+      }
+      i++;
+    }
     for (const part of this._parts) {
-      if (!part) {
-        valueIndex++;
-      } else if (part.size === undefined) {
-        (part as SinglePart).setValue(values[valueIndex]);
-        valueIndex++;
-      } else {
-        (part as MultiPart).setValue(values, valueIndex);
-        valueIndex += part.size;
+      if (part !== undefined) {
+        part.commit();
```

Comment:
Won't the current setup lead to multiple commits for some attributes?

e.g.

```js
const template = (one, two) => html`<my-element some-attr="${one} ${two}"></my-element>`;

render(template('lorem', 'ipsum'), container);

// later on
render(template('dolor', 'sit'), container);
```

Won't that change the `some-attribute` to `"dolor sit"` twice? This loop commits every part, and both `AttributePart`s in the attribute commit their `AttributeCommitter` because they're both dirty.

A possible solution would be to mark all parts of an `AttributeCommitter` as not dirty when running `AttributeCommitter#commit()`

Reply:
This was a mistake, I was switching patterns and forgot to make committing only run once. I'll fix asap :)

Reply:
Fixed and added a test